panic: runtime error in stateVersions.Upload

<!--
Hi there,

Thank you for opening an issue! Please note that we try to keep the this issue
tracker reserved for bug reports and feature requests related to the go-tfe API
wrapper. If you know your issue relates to the Terraform Cloud/Enterprise
platform itself, please contact tf-cloud@hashicorp.support. For general usage
questions, please post to our community forum: https://discuss.hashicorp.com.
-->

#### go-tfe version
<!---
What version of go-tfe are you using?
-->
```plaintext
1.34.0
```

Embedded in Terraform 1.6.1

## Description
When I ran `terraform apply`, terraform dumped with a stacktrace pointing to go-tfe:

```
Terraform has compared your real infrastructure against your configuration
and found no differences, so no changes are needed.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x11e9936]

goroutine 1150 [running]:
github.com/hashicorp/go-tfe.(*stateVersions).Upload.func1()
	/home/runner/go/pkg/mod/github.com/hashicorp/go-tfe@v1.34.0/state_version.go:294 +0xb6
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.3.0/errgroup/errgroup.go:75 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 39
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.3.0/errgroup/errgroup.go:72 +0x96
```

(see https://github.com/golang/sync/blob/v0.3.0/errgroup/errgroup.go#L75 and https://github.com/hashicorp/go-tfe/blob/v1.34.0/state_version.go#L294)

While the log shows the issue in a situation with unchanged infrastructure, this also happens when changes were applied successfully. However, the updated state was not persisted, as I was able to validate with a subsequent terraform plan.

This may or may not be related to the remote backend I am using. In the past, it also sometimes had problems where fields were missing, causing problems with newer versions of go-tfe (via newer versions of terraform).

In any case, I'd have expected a good exception instead of a segfault, which also leaves the terraform state locked instead of failing more gracefully.

The problem happens both in a ci system (linux) as well as on a developer machine (windows 11).

Thank you very much for your support!

## Testing plan
<!--
1.  _Describe how to replicate_
1.  _the conditions under which your code performs its purpose,_
1.  _including example code to run where necessary._
-->
```plaintext
?
```

#### Expected Behavior
go-tfe should not crash

#### Actual Behavior
go-tfe crashed with a segfault

#### Additional Context
<!--
Is there anything atypical about your situation that we should know? For
example: is Terraform running in a wrapper script or in a CI system? Are you
passing any unusual command line options or environment variables to opt-in to
non-default behavior?
-->

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,12 +1,11 @@
-# UNRELEASED
-<!-- Add CHANGELOG entry to this section for any PR awaiting the next release -->
-<!-- Please also include if this is a Bug Fix, Enhancement, or Feature -->
+# v.1.36.0
 
 ## Features
 * Added BETA support for private module registry test variables by @aaabdelgany [#787](https://github.com/hashicorp/go-tfe/pull/787)
 
 ## Bug Fixes
 * Fix incorrect attribute type for `RegistryModule.VCSRepo.Tags` by @hashimoon [#789](https://github.com/hashicorp/go-tfe/pull/789)
+* Fix nil dereference panic within `StateVersions` `upload` after not handling certain state version create errors by @brandonc [#792](https://github.com/hashicorp/go-tfe/pull/792)
 
 # v.1.35.0
 ## Features

--- a/state_version.go
+++ b/state_version.go
@@ -287,6 +287,7 @@ func (s *stateVersions) Upload(ctx context.Context, workspaceID string, options 
 		if strings.Contains(err.Error(), "param is missing or the value is empty: state") {
 			return nil, ErrStateVersionUploadNotSupported
 		}
+		return nil, err
 	}
 
 	g, _ := errgroup.WithContext(ctx)

